src/split-panel/index.tsx

```diff
@@ -32,290 +32,301 @@ export { SplitPanelProps };
-  const cappedSize = getLimitedValue(minSize, size, maxSize);
-  const appLayoutMaxWidth = isRefresh && position === 'bottom' ? contentWidthStyles : undefined;
-
-  const openButtonAriaLabel = i18n('i18nStrings.openButtonAriaLabel', i18nStrings?.openButtonAriaLabel);
-  useEffect(() => {
-    setSplitPanelToggle({ displayed: closeBehavior === 'collapse', ariaLabel: openButtonAriaLabel });
-  }, [setSplitPanelToggle, openButtonAriaLabel, closeBehavior]);
-
-  useEffect(() => {
-    // effects are called inside out in the components tree
-    // wait one frame to allow app-layout to complete its calculations
-    const handle = requestAnimationFrame(() => {
-      const maxSize = position === 'bottom' ? getMaxHeight() : getMaxWidth();
-      setRelativeSize(((size - minSize) / (maxSize - minSize)) * 100);
-      setMaxSize(maxSize);
-    });
-    return () => cancelAnimationFrame(handle);
-  }, [size, minSize, position, getMaxHeight, getMaxWidth]);
-
-  useEffect(() => {
-    reportSize(cappedSize);
-  }, [reportSize, cappedSize]);
-
-  useEffect(() => {
-    const handler = () => setMaxSize(position === 'bottom' ? getMaxHeight() : getMaxWidth());
-    window.addEventListener('resize', handler);
-    return () => window.removeEventListener('resize', handler);
-  }, [position, getMaxWidth, getMaxHeight]);
-
-  const setSidePanelWidth = (width: number) => {
-    const maxWidth = getMaxWidth();
-    const size = getLimitedValue(MIN_WIDTH, width, maxWidth);
-
-    if (isOpen && maxWidth >= MIN_WIDTH) {
-      onResize({ size });
-    }
-  };
-
-  const setBottomPanelHeight = (height: number) => {
-    const maxHeight = getMaxHeight();
-    const size = getLimitedValue(MIN_HEIGHT, height, maxHeight);
+const SplitPanel = forwardRef<SplitPanelProps.Ref, SplitPanelProps>(
+  ({ header, children, hidePreferencesButton = false, closeBehavior = 'collapse', i18nStrings, ...restProps }, ref) => {
+    const isRefresh = useVisualRefresh();
+    const { __internalRootRef } = useBaseComponent('SplitPanel');
+    const {
+      size,
+      getMaxWidth,
+      getMaxHeight,
+      position,
+      topOffset,
+      bottomOffset,
+      rightOffset,
+      contentWidthStyles,
+      isOpen,
+      isForcedPosition,
+      onPreferencesChange,
+      onResize,
+      onToggle,
+      reportSize,
+      setSplitPanelToggle,
+      refs,
+    } = useSplitPanelContext();
+    const baseProps = getBaseProps(restProps);
+    const i18n = useInternalI18n('split-panel');
+    const [isPreferencesOpen, setPreferencesOpen] = useState<boolean>(false);
+    const [relativeSize, setRelativeSize] = useState(0);
+    const [maxSize, setMaxSize] = useState(size);
+    const minSize = position === 'bottom' ? MIN_HEIGHT : MIN_WIDTH;
+    const cappedSize = getLimitedValue(minSize, size, maxSize);
+    const appLayoutMaxWidth = isRefresh && position === 'bottom' ? contentWidthStyles : undefined;
+
+    useImperativeHandle(
+      ref,
+      () => ({
+        focus() {
```

Comment:
Should we call it `focusResizer` to allow adding `focusPanelClose` method without confusion in the future if needed

Reply:
On the other hand, a general `focus` method could be what people should always call when they need to "move focus to the split panel". And if the recommended element to focus changes in the future, we can easily swap out that logic without changing the recommendation.

Reply:
If it would have been unconditional focus on the resizer, we would make it built in bound to the open state.

Since we expose an API instead, it seems like a context-dependent feature, so "just focus" may become limiting